src/components/questionnaire/questionnaire-preview.tsx

```diff
@@ -96,6 +101,14 @@ const QuestionnairePreview = ({
    */
   const handleSaveClick = () => {
     if (editedQuestion !== null && editingIndex !== null) {
+      const allCheckboxesEmpty = editedQuestion.answerOptions.every((option) => !option.isCorrect);
+      const isQuestionTextEmpty = !editedQuestion.questionText.trim();
+
+      if (allCheckboxesEmpty || isQuestionTextEmpty) {
+        alert(`${strings.questionnairePreview.saveAlert}`);
+        return;
+      }
```

Comment:
Could some of this be refactored into a vaidateQuestion function. You could then call the function here and proceed (or not) accordingly. It just cleans up the handleSave function a bit